### PR TITLE
Add hostile/fetch

### LINKED ISSUE #6946: new module: HOSTILE/FETCH
### Is there an existing module for this?

- [x] I have searched for the existing module

### Is there an open PR for this?

- [x] I have searched for existing PRs

### Is there an open issue for this?

- [x] I have searched for existing issues

### Further Information

https://github.com/bede/hostile?tab=readme-ov-file

### Are you going to work on this?

- [ ] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
Closes #6946 

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by jfy133** [comment]
@nf-core-bot fix linting

File: modules/nf-core/hostile/fetch/main.nf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "${task.process}":
+        hostile: \$(hostile --version)
+    END_VERSIONS
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
+    """
+    mkdir reference/
+    export HOSTILE_CACHE_DIR=./reference
```
Comment (LilyAnderssonLee) [inline modules/nf-core/hostile/fetch/main.nf:36]: Is the folder `HOSTILE_CACHE_DIR` default?

**Review by mahesh-panchal — APPROVED** [review]

File: modules/nf-core/hostile/fetch/meta.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      tool_dev_url: "https://github.com/bede/hostile"
+      doi: "10.1093/bioinformatics/btad728"
+      licence: ["MIT"]
+      identifier: biotools:hostile
+
+output:
+  - reference:
+      - "reference/":
+          type: directory
+          description: Directory containing reference required reference genome files
```
Comment (LilyAnderssonLee) [inline modules/nf-core/hostile/fetch/meta.yml:23]: Maybe just `Directory containing reference genome files`?

**Review by LilyAnderssonLee — APPROVED** [review]

File: modules/nf-core/hostile/fetch/main.nf (hunk 1 above)
Comment (jfy133) [reply modules/nf-core/hostile/fetch/main.nf:36]: The default is somewhere in your home directory, which is not accessible in singularity containers and some compute environments

File: modules/nf-core/hostile/fetch/meta.yml (hunk 2 above)
Comment (jfy133) [inline modules/nf-core/hostile/fetch/meta.yml:23]:
```suggestion
          description: Directory containing required reference genome files
```

Comment (jfy133) [reply modules/nf-core/hostile/fetch/meta.yml]: Oops, fixed!

**Review by vagkaratzas — APPROVED** [review]
LGTM

**Pull request merged**